src/pages/Facility/settings/organizations/components/FacilityOrganizationSelector.tsx

```diff
@@ -280,7 +280,17 @@ export default function FacilityOrganizationSelector(
                 variant="ghost"
                 size="sm"
                 className="h-8 gap-2"
-                disabled={alreadySelected}
+                onClick={() => {
+                  if (!currentSelection) return;
+                  const index = selectedOrganizations.findIndex(
+                    (org) => org.id === currentSelection.id,
+                  );
+                  if (index !== -1) {
+                    handleRemoveOrganization(index);
+                  }
+                  setCurrentSelection(null);
```

Comment:
let's move these two inside; we shouldn't have a situation where it fails, but if it does, let's not set current selection to null.